fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -411,6 +411,81 @@ class BackupContainerFileSystemImpl {
 		}
 	}
 
+	// Checks if list of sorted logfiles have the logs from snapshotBeginVersion to snapshotEndversion.
+	// Which means the sorted log files(have beginVersion and endVersion) should cover
+	// all the versions between snapshotBegingVersion and snapshotEndversion.
+	// Note: logs should be pre-sorted according to version order.
+	static bool hasContinuousLogsForSnapshot(const std::vector<LogFile>& logs,
+	                                         Version snapshotBeginVersion,
+	                                         Version snapshotEndVersion) {
+		auto it = logs.begin();
+
+		// find the first mutation log file that covers snapshotBeginVersion
+		while (it != logs.end()) {
+			if (it->beginVersion <= snapshotBeginVersion && it->endVersion > snapshotBeginVersion)
+				break;
+			++it;
+		}
+
+		// no log find found covering snaphostBeginVersion, return false
+		if (it == logs.end())
+			return false;
+
+		// If current log entry(it), covers the entire snapshot, return true
+		if (it->endVersion >= snapshotEndVersion)
```

Comment:
Same as below comments. As `it->endVersion` is an inclusive boundary and snapshotEndVersion is an exclusive boundary, when `it->endVersion == snapshotEndVersion`, the snapshot data at snapshotEndVersion cannot be restored by the current `it` because the current `it` does not have the snapshotEndVersion (due to the exclusive boundary). So, we need to check the next `it`. Please correct me if my understanding is wrong. Thanks!

Reply:
Agree. Thanks!
Changed the code to if `(it->endVersion > snapshotEndVersion)`